crates/parameters/src/directory.rs

```diff
@@ -48,6 +51,7 @@ pub enum SerializationError {
 pub fn deserialize<Parameters>(
     parameters_root_path: impl AsRef<Path>,
     hardware_ids: &Ids,
+    allow_superfluous_fields: bool,
```

Comment:
What do you think about replacing the bool with an enum. To also make this argument readable at call-site?